File: space-game/Core/Classes/DistanceRange.vb
Namespace Classes
    Public Class DistanceRange
        Inherits Range
        Public Sub New(defaultLowerBound As Distance, defaultUpperBound As Distance)
            MyBase.New(defaultLowerBound.Kilometers, defaultUpperBound.Kilometers)
        End Sub

        Public Shadows Sub AddValue(value As Distance)
            MyBase.AddValue(value.Kilometers)
        End Sub

        Public Shadows ReadOnly Property LowerBound As Distance
            Get
                Return Distance.FromKilometers(GetLowerBound())
            End Get
        End Property

        Public Shadows ReadOnly Property UpperBound As Distance
            Get
                Return Distance.FromKilometers(GetUpperBound())
            End Get
        End Property
    End Class
End Namespace

File: space-game/Core/Classes/DistanceRange.cs
namespace Core.Classes
{
    public class DistanceRange : Range
	{
		public DistanceRange(Distance defaultLowerBound, Distance defaultUpperBound) : base(defaultLowerBound.Kilometers, defaultUpperBound.Kilometers)
		{
		}

		public void AddValue(Distance value)
		{
			base.AddValue(value.Kilometers);
		}

		public new Distance LowerBound => Distance.FromKilometers(GetLowerBound());

		public new Distance UpperBound => Distance.FromKilometers(GetUpperBound());
	}
}
